src/api_helpers.jl

```diff
@@ -37,6 +37,13 @@ function h5a_get_name_by_idx(loc_id, obj_name, idx_type, order, idx, lapl_id)
     return String(buf)
 end
 
+function h5a_iterate(f, obj_id, idx_type, order, idx = 0)
+    idxref = Ref{hsize_t}(idx)
+    fptr = @cfunction($f, herr_t, (hid_t, Ptr{Cchar}, Ptr{H5A_info_t}, Ptr{Cvoid}))
```

Comment:
This needs a runtime closure?

Reply:
I'm not sure — I did my best to interpret the documentation, and I thought from the [`@cfunction`](https://docs.julialang.org/en/v1/base/c/#Base.@cfunction) docs saying
```
And that these arguments will be evaluated in global scope during compile-time (not deferred until runtime).
```
that handling the closures would be needed. I can try without.

Reply:
I see it sounds correct then

Reply:
I just confirmed that removing the `$` results in an `f not defined` error get thrown during precompilation.